src/backend/empty/src/descriptor.rs

```diff
@@ -0,0 +1,42 @@
+use hal::pso;
+use log::debug;
+
+/// Dummy descriptor pool.
+#[derive(Debug)]
+pub struct DescriptorPool;
+
+impl pso::DescriptorPool<crate::Backend> for DescriptorPool {
+    unsafe fn allocate_set(
+        &mut self,
+        _layout: &DescriptorSetLayout,
+    ) -> Result<DescriptorSet, pso::AllocationError> {
+        Ok(DescriptorSet {
+            name: String::new(),
+        })
+    }
+
+    unsafe fn free_sets<I>(&mut self, descriptor_sets: I)
+    where
+        I: IntoIterator<Item = DescriptorSet>,
+    {
+        descriptor_sets.into_iter().for_each(std::mem::drop);
```

Comment:
it's a bit weird to call `drop` on the explicitly if we can just iterate them and let them be dropped naturally, i.e.
```rust
for _ in descriptor_sets {}
```